src/box/iproto.cc

```diff
@@ -2107,12 +2120,26 @@ tx_process_select(struct cmsg *m)
 		obuf_rollback_to_svp(out, &svp);
 		goto error;
 	}
-	iproto_reply_select(out, &svp, msg->header.sync,
-			    ::schema_version, count);
+	if (req->fetch_position) {
+		if (iproto_reply_select_with_position(out, &svp,
+						      msg->header.sync,
+						      ::schema_version,
+						      count, packed_pos,
+						      packed_pos_end) != 0) {
+			/* Discard the prepared select. */
+			obuf_rollback_to_svp(out, &svp);
+			goto error;
+		}
+	} else {
+		iproto_reply_select(out, &svp, msg->header.sync,
+				    ::schema_version, count);
```

Comment:
Do we really need `iproto_prepare_select_with_position` when it is the same as `iproto_prepare_select`?

Reply:
I'd make it `static inline` and simply call `iproto_prepare_select` from it. This would clearly delineate the select-without-position vs select-with-position paths in `tx_process_select`.

Reply:
Done